web/src/widgetize.ts

```diff
@@ -1,17 +1,45 @@
 import $ from "jquery";
+import assert from "minimalistic-assert";
 
 import * as blueslip from "./blueslip";
 import * as message_lists from "./message_lists";
+import type {Message} from "./message_store";
 import * as narrow_state from "./narrow_state";
-
-export const widgets = new Map();
-export const widget_contents = new Map();
-
-export function clear_for_testing() {
+import type {Event} from "./poll_widget";
+
+type WidgetOptions = {
+    widget_type: string;
+    extra_data:
+        | {question: string; options: string[]}
+        | {
+              type: string;
+              heading: string;
+              choices: {type: string; reply: string; long_name: string; short_name: string}[];
+          }
+        | undefined;
```

Comment:
Should `undefined` be replaced with `null` ?? .

Reply:
I've some confusion with the type of this `extra_data`. I'll check and get back to this. Thanks!